### PR TITLE
Community/Partners: Add support for Perplexity AI

### PR DESCRIPTION
**Pull request by atherfawaz**
- **Description:** This PR adds support for [Perplexity AI APIs](https://blog.perplexity.ai/blog/introducing-pplx-api). 
  - **Issues:** None
  - **Dependencies:** None
  - **Twitter handle:** [@atherfawaz](https://twitter.com/AtherFawaz)

### CODE REVIEW

**Comment by atherfawaz** [comment]
I don't know how to add/run integration tests. I've reviewed this [doc](https://python.langchain.com/docs/contributing/testing), but need support here. I've _written_ an initial integration test and will continue exploring other partners' integrations meanwhile. 

Also, should this persist in `libs/partners`? Or `lib/community`?

**Comment by atherfawaz** [comment]
So the tests failing due to:
```python
from openai import OpenAI
```
Perplexity is built on top of the openai package. Should I add the `openai` package to some requirements file? I presume Langchain would want to be independent of it. In that case, should I try integrating with the [direct CURL](https://blog.perplexity.ai/blog/introducing-pplx-api#:~:text=curl%20%2DX,%7D%27)?

**Review by baskaryan — COMMENTED** [review]
this is awesome @atherfawaz! could we add some docs notebooks to `docs/docs/integrations/chat` and `docs/docs/integrations/llms`?

**Comment by atherfawaz** [comment]
> this is awesome @atherfawaz! could we add some docs notebooks to `docs/docs/integrations/chat` and `docs/docs/integrations/llms`?

@baskaryan, I've added the notebooks, please have a look. Thanks for fixing the tests! 😄

**Comment by Leewaldoe** [comment]
This is awesome @atherfawaz. From the notebook, it doesn't appear like in-text references are returned. Am I missing something or is this expected?

**Comment by atherfawaz** [comment]
> This is awesome @atherfawaz. From the notebook, it doesn't appear like in-text references are returned. Am I missing something or is this expected?

@Leewaldoe, if by in-text references you mean the source documents aren't returned like they are in Perplexity's web version, then I don't think the API supports it right now -- check these [(1)](https://docs.perplexity.ai/discuss/65c0b02f09d8e3001ca0d3ba) [(2)](https://thenewstack.io/accessing-perplexity-online-llms-programmatically-via-api/). The API is quite limited.

**Comment by atherfawaz** [comment]
@baskaryan, gentle nudge, do you think I should do anything else on this?

File: libs/community/langchain_community/llms/perplexity.py
[the remark this answers is not in the document]
Comment (atherfawaz) [reply libs/community/langchain_community/llms/perplexity.py:1]: What things do we need for an LLM implementation specifically?

**Comment by atherfawaz** [comment]
Hey @baskaryan and @efriis, how should we proceed with this?

File: libs/community/langchain_community/llms/perplexity.py
[the remark this answers is not in the document]
Comment (atherfawaz) [reply libs/community/langchain_community/llms/perplexity.py:1]: @efriis I've removed the LLM and the integration test. Please have a look now.

File: docs/docs/integrations/llms/perplexity.ipynb
Comment (baskaryan) [inline docs/docs/integrations/llms/perplexity.ipynb:1]: not seeing an implementation for a perplexity LLM, am i missing something?

Comment (atherfawaz) [reply docs/docs/integrations/llms/perplexity.ipynb:1]: I was asked to remove it. So I removed it in the last commit

**Review by baskaryan — APPROVED** [review]

**Pull request merged**